src/backend/joanie/core/api/admin.py

```diff
@@ -308,3 +309,32 @@ class ContractDefinitionViewSet(viewsets.ModelViewSet):
     permission_classes = [permissions.IsAdminUser & permissions.DjangoModelPermissions]
     serializer_class = serializers.AdminContractDefinitionSerializer
     queryset = models.ContractDefinition.objects.all()
+
+
+class NestedProductOrderGroupViewSet(
+    NestedGenericViewSet,
+    mixins.CreateModelMixin,
+    mixins.UpdateModelMixin,
+    mixins.RetrieveModelMixin,
+    mixins.ListModelMixin,
+):
+    """
+    OrderGroup ViewSet
+    """
+
+    authentication_classes = [SessionAuthenticationWithAuthenticateHeader]
+    permission_classes = [permissions.IsAdminUser & permissions.DjangoModelPermissions]
+    serializer_class = serializers.AdminOrderGroupSerializer
+    queryset = models.OrderGroup.objects.all().select_related("product")
+    ordering = "created_on"
+    lookup_fields = ["product", "pk"]
+    lookup_url_kwargs = ["product_id", "pk"]
+
+    def get_serializer_class(self):
+        """
+        Return the serializer class to use depending on the action.
+        """
+        if self.action == "create":
+            return serializers.AdminOrderGroupCreateSerializer
+
+        return self.serializer_class
```

Comment:
We may apply this following logic afterwards : 
```suggestion
    def get_serializer_class(self):
        # Use the default serializer if the action is not found in serializer_classes
        return self.serializer_classes.get(self.action, self.default_serializer_class)
```
```

Reply:
Look nice, I take it !